### ISSUE TITLE
plugin-users-permissions gives white screen

### ISSUE THREAD

**Issue by Stavorinus** [strapi/strapi#12384]
### Describe the bug

When I go to Settings ->  USERS & PERMISSIONS PLUGIN -> Roles I get a white screen.

The browser console reports this:

TypeError: undefined is not an object (evaluating '(0,s.default)().allPermissions') — 

Also in when the build runs I noticed this.

warning "@strapi/plugin-users-permissions > @strapi/helper-plugin@4.0.2" has unmet peer dependency "formik@^2.2.6".
133warning "@strapi/plugin-users-permissions > @strapi/helper-plugin@4.0.2" has unmet peer dependency "immer@9.0.6".
134warning "@strapi/plugin-users-permissions > @strapi/helper-plugin@4.0.2" has unmet peer dependency "qs@6.10.1".
135warning "@strapi/plugin-users-permissions > @strapi/helper-plugin@4.0.2" has unmet peer dependency "react-select@^4.0.2".
136warning "@strapi/plugin-users-permissions > grant-koa@5.4.8" has unmet peer dependency "koa@>=2.0.0".
137warning "@strapi/plugin-users-permissions > react-redux@7.2.3" has unmet peer dependency "redux@^2.0.0 || ^3.0.0 || ^4.0.0-0".
138warning "@strapi/plugin-users-permissions > @strapi/helper-plugin > mini-css-extract-plugin@2.4.4" has unmet peer dependency "webpack@^5.0.0".
139warning "@strapi/plugin-users-permissions > @strapi/helper-plugin > styled-components@5.3.3" has unmet peer dependency "react-is@>= 16.8.0".

### System

- Node.js version: v16.13.2
- Strapi version: 4.0.6
- Database: MySQL

**Comment by trondstromlie** [comment]
This bug is also in the latest Strapi version: 4.0.7

**Comment by witek1902** [comment]
Same problem with Strapi 4.0.7
![image](https://user-images.githubusercontent.com/6349150/152371214-d4f5c3a0-a647-4f6d-bc41-d82063eb6f9b.png)

**Comment by derrickmehaffy** [comment]
This was fixed in https://github.com/strapi/strapi/pull/12265

Make sure you delete your package-lock.json or yarn.lock and reinstall node modules and rebuild the admin.

Can't reproduce on v4.0.7 so gonna mark as close until someone can confirm reproduction steps.

**Comment by fxck** [comment]
yea, deleting node_modules and yarn.lock did resolve it

**Comment by derrickmehaffy** [comment]
FYI this step is listed in the update guide:

https://docs.strapi.io/developer-docs/latest/update-migration-guides/update-version.html#upgrade-your-dependencies

![image](https://user-images.githubusercontent.com/8593673/152609427-bec403da-4c4d-4fe9-be80-5333cb72d217.png)

**Comment by SteveScott** [comment]
> FYI this step is listed in the update guide:
> 
> https://docs.strapi.io/developer-docs/latest/update-migration-guides/update-version.html#upgrade-your-dependencies
> 
> ![image](https://user-images.githubusercontent.com/8593673/152609427-bec403da-4c4d-4fe9-be80-5333cb72d217.png)

I was not able to resolve this issue. Steps I took.
-- Remove package-lock.json and yarn.lock
-- remove yarn.lock
-- remove node-modules
-- run yarn install
-- remove yarn.lock again
-- run yarn develop
-- clear browser cache
-- navigate to plugins.

Result:
-- blank screen with above-mentioned error in console.

**Comment by Stavorinus** [comment]
Also fixed for me: 

In my package.json I forgot to upgrade the permission:

   "@strapi/plugin-users-permissions": **"4.0.7"**,

**Comment by julesrenaud** [comment]
@derrickmehaffy I've stumbled into this issue today and wasted a LOT of time before I figured out my issue was having qs as a dependency in my package.json file. It overrode the 6.10.1 version specified in the @strapi/admin package.json file (and other Strapi core packages) with the current version (6.10.3) which broke the admin in the same way documented in this thread. I'm guessing there's a breaking change in 6.10.3 or 6.10.2

**Comment by TonyDeplanque** [comment]
I have the same problem with custom plugin installed in Strapi 4.0.2. The problem would come from the component CheckPagePermissions

**Comment by sitoftonic** [comment]
@julesrenaud Have tried your solution, adding this to my package.json to force 6.10.1:
```
...
"overrides": {
   "qs": "6.10.1"
}
...
```
However, after removing node_modules, package-lock and the build folder (this actions only should be the solution as provided in the docs) and doing npm install again it did not force every dependency in package-lock to have qs at that version, which makes me wonder if I'm doing something wrong: https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides

In any case, after editing the package-lock.json file manually and locking every qs version to 6.10.1 and then running npm install again after removing node_modules folder, etc. And running the application again it keeps me throwing the same error:

```
TypeError: Cannot read properties of undefined (reading 'allPermissions')
    at g (main.586e7c6c.js:2:1337562)
    at li (main.586e7c6c.js:2:1148314)
    at $u (main.586e7c6c.js:2:1200611)
    at Cl (main.586e7c6c.js:2:1187852)
    at Pl (main.586e7c6c.js:2:1187780)
    at jl (main.586e7c6c.js:2:1187643)
    at xl (main.586e7c6c.js:2:1184609)
    at main.586e7c6c.js:2:1134106
    at t.unstable_runWithPriority (main.586e7c6c.js:2:2894254)
    at Go (main.586e7c6c.js:2:1133883)
```

which is being shown in almost every route of my application except for /admin and /content-manager, but this last only on first render, after changing collection it appears another error:
 
```
TypeError: Cannot read properties of undefined (reading 'toggleNotification')
    at t.default (main.586e7c6c.js:2:1797953)
    at d (main.586e7c6c.js:2:1647993)
    at li (main.586e7c6c.js:2:1148314)
    at $u (main.586e7c6c.js:2:1200611)
    at Cl (main.586e7c6c.js:2:1187852)
    at Pl (main.586e7c6c.js:2:1187780)
    at jl (main.586e7c6c.js:2:1187643)
    at xl (main.586e7c6c.js:2:1184609)
    at main.586e7c6c.js:2:1134106
```

I have also tried to downgrade the Strapi version to 4.0.7 for if the link @derrickmehaffy shared has to be taken literally, as it has a reference to that specific version in the example code snippet. However this did not work either.

My package.json is as follows:

```
{
  "name": "obfuscated",
  "private": true,
  "version": "0.1.0",
  "description": "A Strapi application",
  "scripts": {
    "develop": "strapi develop",
    "start": "strapi start",
    "build": "strapi build"
  },
  "dependencies": {
    "@strapi/plugin-documentation": "4.0.7",
    "@strapi/plugin-i18n": "4.0.7",
    "@strapi/plugin-users-permissions": "4.0.7",
    "@strapi/provider-email-mailgun": "4.0.7",
    "@strapi/strapi": "4.0.7",
    "pg": "8.6.0",
    "react-quill": "^1.3.5"
  },
  "overrides": {
    "qs": "6.10.1"
  },
  "author": {
    "name": "A Strapi developer"
  },
  "strapi": {
    "uuid": "whatever-uuid"
  },
  "engines": {
    "node": ">=12.x.x <=16.x.x",
    "npm": ">=6.0.0"
  },
  "license": "MIT"
}
```

**Comment by derrickmehaffy** [comment]
> @julesrenaud Have tried your solution, adding this to my package.json to force 6.10.1:
> 
> ```
> ...
> "overrides": {
>    "qs": "6.10.1"
> }
> ...
> ```
> 
> However, after removing node_modules, package-lock and the build folder (this actions only should be the solution as provided in the docs) and doing npm install again it did not force every dependency in package-lock to have qs at that version, which makes me wonder if I'm doing something wrong: https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides
> 
> In any case, after editing the package-lock.json file manually and locking every qs version to 6.10.1 and then running npm install again after removing node_modules folder, etc. And running the application again it keeps me throwing the same error:
> 
> ```
> TypeError: Cannot read properties of undefined (reading 'allPermissions')
>     at g (main.586e7c6c.js:2:1337562)
>     at li (main.586e7c6c.js:2:1148314)
>     at $u (main.586e7c6c.js:2:1200611)
>     at Cl (main.586e7c6c.js:2:1187852)
>     at Pl (main.586e7c6c.js:2:1187780)
>     at jl (main.586e7c6c.js:2:1187643)
>     at xl (main.586e7c6c.js:2:1184609)
>     at main.586e7c6c.js:2:1134106
>     at t.unstable_runWithPriority (main.586e7c6c.js:2:2894254)
>     at Go (main.586e7c6c.js:2:1133883)
> ```
> 
> which is being shown in almost every route of my application except for /admin and /content-manager, but this last only on first render, after changing collection it appears another error:
> 
> ```
> TypeError: Cannot read properties of undefined (reading 'toggleNotification')
>     at t.default (main.586e7c6c.js:2:1797953)
>     at d (main.586e7c6c.js:2:1647993)
>     at li (main.586e7c6c.js:2:1148314)
>     at $u (main.586e7c6c.js:2:1200611)
>     at Cl (main.586e7c6c.js:2:1187852)
>     at Pl (main.586e7c6c.js:2:1187780)
>     at jl (main.586e7c6c.js:2:1187643)
>     at xl (main.586e7c6c.js:2:1184609)
>     at main.586e7c6c.js:2:1134106
> ```
> 
> I have also tried to downgrade the Strapi version to 4.0.7 for if the link @derrickmehaffy shared has to be taken literally, as it has a reference to that specific version in the example code snippet. However this did not work either.
> 
> My package.json is as follows:
> 
> ```
> {
>   "name": "obfuscated",
>   "private": true,
>   "version": "0.1.0",
>   "description": "A Strapi application",
>   "scripts": {
>     "develop": "strapi develop",
>     "start": "strapi start",
>     "build": "strapi build"
>   },
>   "dependencies": {
>     "@strapi/plugin-documentation": "4.0.7",
>     "@strapi/plugin-i18n": "4.0.7",
>     "@strapi/plugin-users-permissions": "4.0.7",
>     "@strapi/provider-email-mailgun": "4.0.7",
>     "@strapi/strapi": "4.0.7",
>     "pg": "8.6.0",
>     "react-quill": "^1.3.5"
>   },
>   "overrides": {
>     "qs": "6.10.1"
>   },
>   "author": {
>     "name": "A Strapi developer"
>   },
>   "strapi": {
>     "uuid": "whatever-uuid"
>   },
>   "engines": {
>     "node": ">=12.x.x <=16.x.x",
>     "npm": ">=6.0.0"
>   },
>   "license": "MIT"
> }
> ```

Simple suggestion is to delete your `node_modules` and lock files and reinstall node_modules

**Comment by sitoftonic** [comment]
> Simple suggestion is to delete your `node_modules` and lock files and reinstall node_modules
@derrickmehaffy This is what I have done several times without any signal of improvement. Is there something else I can do to debug the problem?

**Comment by derrickmehaffy** [comment]
> > Simple suggestion is to delete your `node_modules` and lock files and reinstall node_modules
> 
> @derrickmehaffy This is what I have done several times without any signal of improvement. Is there something else I can do to debug the problem?
> 
> **Note**: Cannot understand why this ticket is being closed, unfortunately this is happening to me (and I can imagine to other people too) and the proposed solution is not working at all...

Because we cannot reproduce it and everyone who I've helped on our discord fixed it by doing exactly what I've said before:

Delete `build`, `.cache`, `node_modules`, `yarn.lock`, package-lock.json` then reinstall your node_modules (and optionally rebuilding the admin). That all permissions error is 100% the fault of a cached admin build. If those steps don't work, it's likely your system has a hard cache for either npm or yarn.

The only other option is you have customized the admin panel using patch-package or something that we can't help with.

**Comment by Nikoxx99** [comment]
> @derrickmehaffy After some days stuck in this problem I have managed to solve it. First I tried removing the .cache folder, as well as node_modules, package-lock.json and build folder. Also I cleared npm cache using:
> 
> ```
> npm cache clean --force
> ```
> 
> In my case I did not customize anything from the admin panel, my case is a simple Strapi installation with a few plugins installed (package.json can be found in my first comment in this thread).
> 
> The solution: Just as simple as moving from npm to yarn. Once I removed package-lock.json, set up yarn in my system:
> 
> ```
> npm i -g yarn
> ```
> 
> and running:
> 
> ```
> yarn
> ```
> 
> and then rebuilding the admin panel:
> 
> ```
> npm run build
> ```
> 
> And now it is working! I still keep wondering why the error keep happening after clearing npm's cache...

Also try installing dependencies with yarn and it fix the error. Npm is doing some weird things...

**Comment by derrickmehaffy** [comment]
Lol yup, no idea why but yarn > npm

**Comment by b3nab** [comment]
As far as I know npm resolves dependencies in a different way than yarn. That's why there are differences running it using npm vs yarn.
But I think that npm sticks more to what is declared inside dependencies' package.json instead yarn will "overwrite" some dependencies. There are pros and cons.

**Anyway it could be interesting comparing the two lock files, so find out differences between package-lock.json and yarn.lock that are breaking things.**

I found out that there are two main issues, 1 is that I need to stick to react 17 otherwise some plugins didn't work well and the admin is giving white pages all the time, even with yarn and 2 is that I had to switch to yarn because the same problem on this thread.
